experimental/client.go

```diff
@@ -0,0 +1,338 @@
+		policyTemplate, ok := policyTemplates[i].(map[string]interface{})
+		if !ok {
+			fmt.Fprintf(os.Stderr, "An invalid policy-templates entry was provided: %v\n", err)
+			os.Exit(1)
+		}
+
+		objectDefinition, ok := policyTemplate["objectDefinition"].(map[string]interface{})
+		if !ok {
+			fmt.Fprintf(os.Stderr, "An invalid policy-templates entry was provided: %v\n", err)
+			os.Exit(1)
+		}
+
+		objectDefinitionUnstructured := unstructured.Unstructured{Object: objectDefinition}
+		if objectDefinitionUnstructured.GetAPIVersion() != "policy.open-cluster-management.io/v1" {
+			continue
+		}
+
+		if objectDefinitionUnstructured.GetKind() != "ConfigurationPolicy" {
+			continue
+		}
+
+		if hubResolver != nil {
+			objectDefinitionJSON, err := json.Marshal(objectDefinition)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "An invalid policy-templates entry at index %d was provided: %v\n", i, err)
+				os.Exit(1)
+			}
+
+			hubTemplateResult, err := hubResolver.ResolveTemplate(objectDefinitionJSON, hubTemplateCtx)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "An invalid policy-templates entry at index %d was provided: %v\n", i, err)
+				os.Exit(1)
+			}
+
+			var resolvedObjectDefinition map[string]interface{}
+
+			err = json.Unmarshal(hubTemplateResult.ResolvedJSON, &resolvedObjectDefinition)
+			if err != nil {
+				fmt.Fprintf(
+					os.Stderr,
+					"An invalid policy-templates entry at index %d after resolving templates: %v\n",
+					i,
+					err,
+				)
+				os.Exit(1)
+			}
+
+			err = unstructured.SetNestedField(policyTemplate, resolvedObjectDefinition, "objectDefinition")
+			if err != nil {
+				fmt.Fprintf(
+					os.Stderr,
+					"An invalid policy-templates entry at index %d after resolving templates: %v\n",
+					i,
+					err,
+				)
+				os.Exit(1)
+			}
+
+			objectDefinition = policyTemplate["objectDefinition"].(map[string]interface{})
+			objectDefinitionUnstructured.Object = objectDefinition
+		}
+
+		var rawDataList [][]byte
+
+		oTRaw, oTRawFound, _ := unstructured.NestedString(objectDefinition, "spec", "object-templates-raw")
+		if oTRawFound {
+			rawDataList = [][]byte{[]byte(oTRaw)}
+		} else {
+			objTemplates, _, err := unstructured.NestedSlice(objectDefinition, "spec", "object-templates")
+			if err != nil {
+				fmt.Fprintf(
+					os.Stderr,
+					"The ConfigurationPolicy at policy-templates index %d has an invalid object-templates array: %v\n",
+					i,
+					err,
+				)
+				os.Exit(1)
+			}
+
+			for _, objTemplate := range objTemplates {
+				jsonBytes, err := json.Marshal(objTemplate)
+				if err != nil {
+					fmt.Fprintf(
+						os.Stderr,
+						"The ConfigurationPolicy at policy-templates index %d has an invalid object-templates "+
+							"array: %v\n",
+						i,
+						err,
+					)
+					os.Exit(1)
+				}
+
+				rawDataList = append(rawDataList, jsonBytes)
+			}
+		}
+
+		objectTemplates := make([]interface{}, 0, len(rawDataList))
+
+		for _, rawData := range rawDataList {
```

Comment:
Not sure how feasible/easy this would be, but I might like it if the logic in this loop were broken out into a function so that an argument like `-raw` could call that function directly on my file containing only an `object-templates-raw` value. Then some default values could be filled in for the rest of the Policy stuffs so I wouldn't have to think about that and I could get down to business with a big ol' template.

Reply:
RFE deferred :smile: